### DISCUSSION TITLE
resource_dir() invalid expected result on windows

### DISCUSSION THREAD

**Discussion by olexiyb** [tauri-apps/tauri#10159] [Ideas]
All, my [pull request](https://github.com/tauri-apps/tauri/pull/10093) has been closed without discussion
I'm afraid I have to disagree with the reason it was closed 

The unit test I wrote contains proof of my point.

```
  #[test]
  fn test_current_exe() {
    let current = super::current_exe().unwrap();
    let current_expected = std::env::current_exe().unwrap();
    assert_eq!(current, current_expected);
  }
  ```
  
  Tauri solved [the security flaw](https://doc.rust-lang.org/std/env/fn.current_exe.html#security) of `std::env::current_exe` but at the same time it does not properly work on Windows as `super::current_exe()` gives different from `std::env::current_exe()` value

**Comment by FabianLars** [comment]
> but at the same time it does not properly work on Windows as super::current_exe() gives different from std::env::current_exe() value

We never said it would work the same, instead we even documented that it uses the std canonicalize function (which introduces the problem).

**Comment by olexiyb** [comment]
At this moment

```
pub fn current_exe() -> std::io::Result<PathBuf> {
  self::starting_binary::STARTING_BINARY.cloned()
}
```

Where 

```
#[ctor]
#[used]
pub(super) static STARTING_BINARY: StartingBinary = StartingBinary::new();

/// Represents a binary path that was cached when the program was loaded.
pub(super) struct StartingBinary(std::io::Result<PathBuf>);

impl StartingBinary {
  /// Find the starting executable as safely as possible.
  fn new() -> Self {
    // see notes on current_exe() for security implications
    let dangerous_path = match `std::env::current_exe()` {
      Ok(dangerous_path) => dangerous_path,
      error @ Err(_) => return Self(error),
    };

    // note: this only checks symlinks on problematic platforms, see implementation below
    if let Some(symlink) = Self::has_symlink(&dangerous_path) {
      return Self(Err(Error::new(
        ErrorKind::InvalidData,
        format!("StartingBinary found current_exe() that contains a symlink on a non-allowed platform: {}", symlink.display()),
      )));
    }

    // we canonicalize the path to resolve any symlinks to the real exe path
    Self(dangerous_path.canonicalize())
  }
```

Which calls `std::env::current_exe` inside, so why unit test on windows when I call `std::env::current_exe` does not equal to call of ctored `current_exe()` which is just a wrapper around `std::env::current_exe`?
